tools/releaser/main.go

```diff
@@ -0,0 +1,168 @@
+// Copyright 2023 Uber Technologies, Inc.
+// Licensed under the Apache License, Version 2.0
+
+// releaser is a tool for managing part of the process to release a new version of bazel-zig-cc.
+package main
+
+import (
+	"bytes"
+	"compress/gzip"
+	"crypto/sha256"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path"
+	"regexp"
+	"strings"
+)
+
+var (
+	// Paths to be included to the release
+	_paths = []string{
+		"LICENSE",
+		"NOTICE",
+		"README.md",
+		"toolchain/*",
+	}
+
+	// regexp for valid tags
+	tagRegexp = regexp.MustCompile(`^v([0-9]+)\.([0-9]+)(\.([0-9]+))(-rc([0-9]+))?$`)
+
+	errTag = errors.New("tag accepts the following formats: v1.0.0 v1.0.1-rc1")
+)
+
+func main() {
+	if err := run(); err != nil {
+		fmt.Fprintf(os.Stderr, "error: %s\n", err)
+		os.Exit(1)
+	}
+}
+
+func log(msg string, format ...any) {
+	fmt.Fprintf(flag.CommandLine.Output(), msg+"\n", format...)
+}
+
+func run() (_err error) {
+	var (
+		repoRoot string
+		tag      string
+	)
+
+	flag.StringVar(&repoRoot, "repo_root", os.Getenv("BUILD_WORKSPACE_DIRECTORY"), "root directory of bazel-zig-cc repo")
+	flag.StringVar(&tag, "tag", "", "tag for this release")
+
+	flag.Usage = func() {
+		fmt.Fprint(flag.CommandLine.Output(), `usage: bazel run //tools/releaser -- -go_version <version> -tag <tag>
+
+This utility is intended to handle many of the steps to release a new version.
+
+`)
+		flag.PrintDefaults()
+	}
+
+	flag.Parse()
+
+	if tag == "" {
+		return fmt.Errorf("tag is required")
+	}
+
+	if !tagRegexp.MatchString(tag) {
+		return errTag
+	}
+
+	// commands that Must Not Fail
+	cmds := [][]string{
+		{"git", "diff", "--stat", "--exit-code"},
+		{"git", "tag", tag},
+	}
+
+	log("Cutting a release:")
+
+	for _, c := range cmds {
+		cmd := exec.Command(c[0], c[1:]...)
+		cmd.Dir = repoRoot
+		if out, err := cmd.CombinedOutput(); err != nil {
+			return fmt.Errorf(
+				"run %s: %w\n%s",
+				strings.Join(c, " "),
+				err,
```

Comment:
Can you convert this `err` to exitErr so we can see the stderr?

Reply:
it is `CombinedOutput`, so we're good.